src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs

```diff
@@ -504,21 +504,22 @@ internal MigrationAssessmentInfo[] ParseAssessmentResult(IList<ISqlMigrationAsse
         {
             return assessmentResults.Select(r =>
             {
+                var check = (Microsoft.SqlServer.Management.Assessment.Checks.Check)r.Check;
                 return new MigrationAssessmentInfo()
                 {
-                    CheckId = r.Check.Id,
-                    Description = r.Check.Description,
-                    DisplayName = r.Check.DisplayName,
-                    HelpLink = r.Check.HelpLink,
-                    Level = r.Check.Level.ToString(),
+                    CheckId = check.Id,
+                    Description = check.Description,
+                    DisplayName = r.Message,
+                    HelpLink = check.HelpLink,
+                    Level = check.Level.ToString(),
                     TargetType = r.TargetType.ToString(),
                     DatabaseName = r.DatabaseName,
                     ServerName = r.ServerName,
-                    Tags = r.Check.Tags.ToArray(),
+                    Tags = check.Tags.ToArray(),
```

Comment:
should there be null checks for check.Level, r.TargetType, or check.Tags, can they be null?

Reply:
`check.Level` and `r.TargetType` cannot be null as they are fundamental to the migration assessment info. `check.Tags` can be empty but it won't be null. The NuGet provides a complete result regardless, this is just mapping the NuGet result to what ADS expects. Since most of these fields are unused anyways, prior to this change (where there never were any null checks to begin with) I hadn't seen any issues.